pkgs/applications/misc/nwg-panel/default.nix

```diff
@@ -0,0 +1,68 @@
+{ lib, fetchFromGitHub
+, python3Packages, wrapGAppsHook, gobject-introspection
+, gtk-layer-shell, pango, gdk-pixbuf, atk
+# Extra packages called by various internal nwg-panel modules
+, sway             # swaylock, swaymsg
+, systemd          # systemctl
+, wlr-randr        # wlr-randr
+, nwg-menu
+, withBluetoothControl ? true
+, withBrightnessControl ? true, light
+, withAudioControl ? true, pamixer, pulseaudio
```

Comment:
Do we even need those options?

Reply:
> Do we even need those options?

No, this is mostly to reduce the closure size if needs be.

I wasn't sure whether for tools that rely on many programs we should add all the packages to the closure.

Reply:
Though pulseaudio is most likely already part of most users environment.
As for the other programs, they are small and have few dependencies.

I'll remove these things and make them mandatory.

Further, `makeBinPath` ignores `null` packages so it would still be possible to set them to null.

Reply:
Personally I think leaving them is a good idea, but leaving them default is the way to go.

Reply:
> Further, makeBinPath ignores null packages so it would still be possible to set them to null.

Overwriting random packages to null is most of the time unsupported and can result in undefined behaviour.

Reply:
> Options for a smaller closure size is always good in my book.

Depends. For packages like diffoscope with lots of very big dependencies this makes sense but that does not really apply here. Who will be using this package without the tiny light package? The difference in closure size is negligible and having those options just makes the package unnecessary complex.

Reply:
> > Further, makeBinPath ignores null packages so it would still be possible to set them to null.
> 
> Overwriting random packages to null is most of the time unsupported and can result in undefined behaviour.

I meant that in the sense of `nwg-panel.override { thingy = null; }` or the equivalent `callPackage`.

Reply:
> nwg-panel.override { thingy = null; }

You can do that but that may result in eval erorrs or build failures.